common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/PostgresqlClient.java

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright (c) 2021 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql;
+
+import org.eclipse.dataspaceconnector.clients.postgresql.row.RowMapper;
+
+import java.sql.SQLException;
+import java.util.List;
+
+public interface PostgresqlClient extends AutoCloseable {
+
+    /**
+     * Intended for mutating queries.
+     *
+     * @param sql       the parametrized sql query
+     * @param arguments the parameteres to interpolate with the parametrized sql query
+     * @return rowsChanged
+     * @throws SQLException if execution of the query was failing
+     */
+    int execute(String sql, Object... arguments) throws SQLException;
```

Comment:
Personally I'm not much into checked exception handling, I would prefer returning an object that encapsulate the result or error.

Reply:
Nah, won't introduce an error wrapping return object, since `SQLException` are an integral part when talking to SQL database. Also IMO it would make less readable.